desc/protoparse/parser.go

```diff
@@ -257,6 +270,46 @@ func (p Parser) ParseFilesButDoNotLink(filenames ...string) ([]*dpb.FileDescript
 	return fds, nil
 }
 
+func newImportResolver(importPaths []string) func(string) (string, error) {
+	return func(name string) (string, error) {
+		if !filepath.IsAbs(name) {
+			return name, nil
+		}
+		var currentImportPath string
+		var resolvedPath string
+		var err error
+		for _, importPath := range importPaths {
+			// If this is within this directory, we have a match.
+			if strings.HasPrefix(filepath.ToSlash(name), filepath.ToSlash(importPath)) {
+				// We have a duplicate.
+				if currentImportPath != "" {
+					return "", fmt.Errorf("%s resides in both %s and %s", name, importPath, currentImportPath)
```

Comment:
`protoc` does not produce an error for this case -- it just uses the first matching import path. So let's not do this.